apps/src/code-studio/components/ModelManagerDialog.jsx

```diff
@@ -104,9 +164,52 @@ export default class ModelManagerDialog extends React.Component {
               isPending={this.state.isImportPending}
               pendingText={'Importing...'}
             />
+            {showDeleteButton && (
+              <Button
+                text={'Delete'}
+                color={Button.ButtonColor.red}
+                onClick={this.showDeleteConfirmation}
+                disabled={noModels}
+                icon={'trash'}
+                iconClassName={'fa-trash'}
+              />
+            )}
+          </div>
+          <div style={styles.right}>
+            <ModelCard model={this.state.selectedModel} />
+          </div>
+        </BaseDialog>
+        <BaseDialog
+          isOpen={this.state.confirmDialogOpen}
+          handleClose={this.closeConfirmDialog}
+          useUpdatedStyles
+          style={styles.dialog}
+        >
+          <h1 style={styles.header}>
+            Are you sure you would like to delete this model?
+          </h1>
+          <div style={styles.left}>
+            <p style={styles.message}>
+              This model will be permanently deleted, and you will not be able
+              to use this model in any App Lab projects.
+            </p>
+            <div>
+              <Button
+                text={'No'}
+                color={Button.ButtonColor.orange}
+                onClick={this.closeConfirmDialog}
```

Comment:
Pressuming "No" closes the entire dialog, but I think it should actually go back to the previous dialog state in which you can choose to import/delete.